### ISSUE TITLE
Improvement on managing MaskDetailer when the bbox detector does not detect anything

### ISSUE THREAD

**Issue by yoda5477** [ltdrdata/ComfyUI-Impact-Pack#363]
Currenly if no mask is provided to MaskDetailer, it creates error and stop workflow. Would suggest to simply ignore and push the input imlage toward the output unchanged.

(Did this dirty below:

       if mask != None:
        # create segs
            mask = make_2d_mask(mask)

            segs = core.mask_to_segs(mask, False, crop_factor, False, drop_size)

        enhanced_img_batch = None
        cropped_enhanced_list = []
        cropped_enhanced_alpha_list = []

        for i in range(batch_size):
            if mask != None:
                enhanced_img, _, cropped_enhanced, cropped_enhanced_alpha, _, new_segs = \
                    DetailerForEach.do_detail(image, segs, model, clip, vae, guide_size, guide_size_for, max_size, seed+i, steps,
                                          cfg, sampler_name, scheduler, positive, negative, denoise, feather, mask_mode,
                                          force_inpaint=True, wildcard_opt=None, detailer_hook=detailer_hook,
                                          refiner_ratio=refiner_ratio, refiner_model=refiner_model, refiner_clip=refiner_clip,
                                          refiner_positive=refiner_positive, refiner_negative=refiner_negative, cycle=cycle)
            else:
                enhanced_img, _, cropped_enhanced, cropped_enhanced_alpha, _, new_segs = image,None,None,None,None,None
                
            if enhanced_img_batch is None:
                enhanced_img_batch = enhanced_img
            else:
                enhanced_img_batch = torch.cat((enhanced_img_batch, enhanced_img), dim=0)
        
            if mask != None:
                cropped_enhanced_list += cropped_enhanced
                cropped_enhanced_alpha_list += cropped_enhanced_alpha
)

**Comment by ltdrdata** [comment]
This is fixed on V4.46.2
